modules/javafx.graphics/src/main/java/com/sun/javafx/iio/javax/XImageLoader.java

```diff
@@ -152,53 +153,66 @@ public ImageFrame load(int imageIndex, double w, double h, boolean preserveAspec
                     image.getWidth(), image.getHeight(), image.getWidth() * 4,
-                    null, pixelScale, metadata);
+                    pixelScale, metadata);
 
             case TYPE_4BYTE_ABGR_PRE -> new ImageFrame(ImageStorage.ImageType.ABGR_PRE,
-                    getByteBuffer(image.getData().getDataBuffer()),
+                    getByteBuffer(image.getRaster().getDataBuffer()),
                     image.getWidth(), image.getHeight(), image.getWidth() * 4,
-                    null, pixelScale, metadata);
+                    pixelScale, metadata);
 
             case TYPE_INT_RGB -> new ImageFrame(ImageStorage.ImageType.INT_RGB,
-                    getIntBuffer(image.getData().getDataBuffer()),
+                    getIntBuffer(image.getRaster().getDataBuffer()),
                     image.getWidth(), image.getHeight(), image.getWidth() * 4,
-                    null, pixelScale, metadata);
+                    pixelScale, metadata);
 
             case TYPE_INT_BGR -> new ImageFrame(ImageStorage.ImageType.INT_BGR,
-                    getIntBuffer(image.getData().getDataBuffer()),
+                    getIntBuffer(image.getRaster().getDataBuffer()),
                     image.getWidth(), image.getHeight(), image.getWidth() * 4,
-                    null, pixelScale, metadata);
+                    pixelScale, metadata);
 
             case TYPE_INT_ARGB -> new ImageFrame(ImageStorage.ImageType.INT_ARGB,
-                    getIntBuffer(image.getData().getDataBuffer()),
+                    getIntBuffer(image.getRaster().getDataBuffer()),
                     image.getWidth(), image.getHeight(), image.getWidth() * 4,
-                    null, pixelScale, metadata);
+                    pixelScale, metadata);
 
             case TYPE_INT_ARGB_PRE -> new ImageFrame(ImageStorage.ImageType.INT_ARGB_PRE,
-                    getIntBuffer(image.getData().getDataBuffer()),
+                    getIntBuffer(image.getRaster().getDataBuffer()),
                     image.getWidth(), image.getHeight(), image.getWidth() * 4,
-                    null, pixelScale, metadata);
+                    pixelScale, metadata);
+
+            case TYPE_BYTE_BINARY, TYPE_BYTE_INDEXED -> {
+                var colorModel = (IndexColorModel)image.getColorModel();
+                var palette = new int[colorModel.getMapSize()];
+                colorModel.getRGBs(palette);
+                yield new ImageFrame(
+                    colorModel.hasAlpha()
+                        ? colorModel.isAlphaPremultiplied()
+                            ? ImageStorage.ImageType.PALETTE_ALPHA_PRE
+                            : ImageStorage.ImageType.PALETTE_ALPHA
+                        : ImageStorage.ImageType.PALETTE,
+                    getByteBuffer(image.getRaster().getDataBuffer()),
+                    image.getWidth(), image.getHeight(), image.getWidth() * colorModel.getPixelSize(),
```

Comment:
General comment here for all spots where you are calculating the stride; you can ask AWT this.  You don't know what kind of padding or rounding going on, and given how varied image formats can be, its unlikely to be always as simple as "round up to nearest byte".  I also think that having in some cases the stride being bytes and in others being bits is not a good idea.  So, how often before the switch asking AWT for the stride?

      SampleModel model = raster.getSampleModel();

      int stride = switch(model) {
          case ComponentSampleModel m -> m.getScanlineStride();
          case MultiPixelPackedSampleModel m -> m.getScanlineStride();
          case SinglePixelPackedSampleModel m -> m.getScanlineStride();
          default -> throw new IllegalStateException("Unsupported sample model: " + model);
      }

(Blame AWT for not putting `getScanlineStride` as an abstract method in the base class).

You can then use this value for all locations, and IMHO should convert the indexed versions to use bytes, or at a minimum document in `ImageFrame` that `stride` no longer has its traditional meaning in all cases.  As I think that's very confusing, and a stride in bits is useless to have, I'd be very much in favor of having the stride back to being bytes in all cases.

Reply:
Implemented as you suggested. Scanline stride is now also always measured in bytes.